autometa/taxonomy/ncbi.py

```diff
@@ -338,57 +332,38 @@ def parse_merged(self):
             logger.debug('merged loaded')
         return merged
 
-    def get_lineage_names(self, taxid):
-        """
-        Returns the lineage names of the taxid
-
-        Parameters
-        ----------
-        taxid : int
-            identifer for a taxon in the Taxonomy Database by NCBI
-
-        Returns
-        -------
-        set
-            lineage of the taxid
-        """
-        lineage = set()
-        while taxid != 1:
-            if self.rank(taxid) not in NCBI.CANONICAL_RANKS:
-                taxid = self.parent(taxid)
-                continue
-            lineage.add(self.name(taxid))
-            taxid = self.parent(taxid)
-        return lineage
-
-    def is_common_ancestor(self, parent_taxid, child_taxid):
+    def is_common_ancestor(self, taxid1, taxid2):
         """
         Determines whether the provided taxids have a non-root common ancestor
 
         Parameters
         ----------
-        parent_taxid : int
+        taxid1 : int
             identifer for a taxon in the Taxonomy Database by NCBI
-        child_taxid : int
+        taxid2 : int
             identifer for a taxon in the Taxonomy Database by NCBI
 
         Returns
         -------
         boolean
             True if taxids share a common ancestor else False
         """
-        self.is_valid_taxid(parent_taxid)
-        self.is_valid_taxid(child_taxid)
-        parent_lineage = self.get_lineage_names(parent_taxid)
-        child_lineage = self.get_lineage_names(child_taxid)
-        common_ancestor = child_lineage.intersection(parent_lineage)
-        if common_ancestor:
-            return True
-        return False
+        taxid1_lineage = self.lineage(taxid1)
+        taxid2_lineage = self.lineage(taxid2)
+        taxid1_lineage_taxids = set([taxids.get('taxid')
+                                     for taxids in taxid1_lineage])
+        taxid2_lineage_taxids = set([taxids.get('taxid')
+                                     for taxids in taxid2_lineage])
```

Comment:
Where are `taxids` coming from?

Reply:
Sorry, I am not quite sure what you mean here

Reply:
Ah wait, I see. Nevermind, missed the list comprehension.